daemon/cmd/hubble.go

```diff
@@ -322,6 +323,8 @@ func (d *Daemon) launchHubble() {
 			}
 		}()
 	}
+
+	d.hubbleObserver.Store(hubbleObserver)
```

Comment:
One functional difference here is that now we are setting the observer only if all the previous operations succeeded, while it was always configured in the previous implementation. I'd defer the final call on whether this is acceptable to Alexandre though, as I don't know the details of the hubble initialization logic.

Reply:
Good point, thanks. I suspect that doesn't really make a difference in practice. But using defer sounds like a good idea

Reply:
From the commit message:

> Fixing the data race by simply waiting to start the status collector until hubble is initialized does not work, as launchHubble could take up to 30s to return while waiting for a TLS certificate and we don't want to block agent startup for this.

Actually `lauchHubble` can block indefinitely waiting on the TLS cert, the 30s timeout is only used to log the "Wait on TLS cert" message. See

https://github.com/cilium/cilium/blob/39dce8574d453b6957a85923ec768ded07b4ff42/daemon/cmd/hubble.go#L285-L294

While we wait on the TLS cert, Hubble is in a "half initialized state" where metrics and the local UNIX domain socket Hubble server are fully initialized, but the TCP server isn't.

At that point, before this patch we would report `models.StatusStateOk` (because `d.hubbleObserver` is set before waiting on the TLS cert) and after this patch we report `models.HubbleStatusStateWarning` "Server not initialized".

Personally speaking, "Server not initialized" make sense when Hubble is configured to listen on TCP but doesn't yet, so this change LGTM. cc @rolinh @gandro thoughts?

Reply:
> At that point, before this patch we would report models.StatusStateOk (because d.hubbleObserver is set before waiting on the TLS cert) and after this patch we report models.HubbleStatusStateWarning "Server not initialized".
>
> Personally speaking, "Server not initialized" make sense when Hubble is configured to listen on TCP but doesn't yet, so this change LGTM.

That's a good point. I think the change is actually for the better.